### ISSUE TITLE
Ability to prepend/suffix a file when created?

### ISSUE THREAD

**Issue by BlackConure** [gabime/spdlog#54]
I am looking to create a log file in an xml or json format, but this requires some file header and footer information.   

To do this, I need to add the header only once when the file is created and then put in the footer as the file is closed (but also remove it if it was there previously, so that it appends new entries properly).   Is this something that can be added?

**Comment by gabime** [comment]
You can implement yout own custom sink for this. Shoidl be quite easy.

**Comment by BlackConure** [comment]
Yes, perhaps a sink would work.   Though it would be nice to have it part of the feature set because it is fairly common for many types of files.   Without it would require a sink to be written for each one.   It was just an idea I thought might be worth having in the base code.   i.e.

set_file_header("");
set_file_footer("");
